packages/strapi-helper-plugin/lib/src/utils/Manager.js

```diff
@@ -0,0 +1,245 @@
+const { findIndex, pullAt, range } = require('lodash');
+const { List } = require('immutable');
+
+class Manager {
+  constructor(state, list, keys, index, layout) {
+    this.state = state;
+    this.keys = keys.split('.');
+    this.layout = layout;
+    this.list = list;
+    this.index = index;
+    this.arrayOfEndLineElements = this.getLinesBound();
+    this.attrToRemoveInfos = this.attrToRemoveInfos();
+  }
+
+  /**
+   * Retrieve the bootstrap col index, name and type of a field
+   * @param {Number} index 
+   * @returns {Object}
+   */
+  getAttrInfos(index) {
+    const name = this.getAttrName(index);
+    const appearance = this.layout.getIn([name, 'appearance']);
+    const type = appearance !== '' && appearance !== undefined ? appearance : this.getType(name);
+    const bootstrapCol = this.getBootStrapCol(type);
+
+    const infos = {
+      bootstrapCol,
+      index,
+      name,
+      type,
+    };
+
+    return infos;
+  }
+
+  getColsToAdd(number) {
+    let ret;
+
+    switch(number) {
+      case 12:
+        ret = [];
+        break;
+      case 9:
+        ret = ['__col-md-3__', '__col-md-6__'];
+        break;
+      case 8:
+        ret = ['__col-md-4__', '__col-md-4__'];
+        break;
+      case 4:
+        ret = ['__col-md-4__'];
+        break;
+      case 6:
+        ret = ['__col-md-6__'];
+        break;
+      default:
+        ret = ['__col-md-3__'];
+    }
+
+    const random = Math.random().toString(36).substring(7);
+    const random1 = Math.random().toString(36).substring(8);
+
+    return ret.map((v, i) => {
+
+      if (i === 0) {
+        return `${v}${random}`;
+      }
+
+      return `${v}${random1}`;
+    });
+  }
+
+  /**
+   * Retrieve a field default bootstrap col
+   * NOTE: will change if we add the customisation of an input's width
+   * @param {String} type 
+   * @returns {Number}
+   */
+  getBootStrapCol(type) {
+    switch(type) {
+      case 'checkbox':
+      case 'boolean':
+      case 'date':
+      case 'bigint':
+      case 'decimal':
+      case 'float':
+      case 'integer':
+      case 'number':
+        return 4;
+      case 'json':
+      case 'wysiwyg':
+      case 'WYSIWYG':
+        return 12;
+      default:
+        return 6;
+    }
+  }
+
+  getElementsOnALine(itemsToPull, arr = this.list) {
+    const array = List.isList(arr) ? arr.toJS() : arr;
+
+    return pullAt(array, itemsToPull);
+  }
+
+  /**
+   * Retrieve the field to remove infos
+   * @returns {Object}
+   */
+  attrToRemoveInfos() {
+    return this.getAttrInfos(this.index);
+  }
+
+  /**
+   * 
+   * Retrieve the last element of each bootstrap line
+   * @returns {Array}
+   */
+  getLinesBound() { // NOTE: doesn't work for the last element if the line is not full!
+    const array = [];
+    let sum = 0;
```

Comment:
What's `sum`? I would prefer a more comprehensive name. The algorithm is complex and even more, if the variables aren't named correctly